sys/sys/uio.h

```diff
@@ -78,6 +78,7 @@ struct vm_object;
 struct vm_page;
 struct bus_dma_segment;
 
+struct uio *alloc_uio(u_int iovcnt);
```

Comment:
Do we want a corresponding function to free uios? That'd require some churn but it's not *too* bad I think? I'd support upstreaming the allocuio() wrapper at least.

Reply:
If we go the UMA way, we can probably avoid both allocuio and freeuio by having a ctor/dtor. We would need the ctor anyway to handle the embedded iovec array anyway. I think we would maintain M_IOV and use it also for large iovec arrays allocations (a bit like mbufs' `m_data` I guess)

Reply:
That could work, though you might want to have some heuristic to ensure that very large iovecs get freed promptly instead of lingering in the UMA cache. (This could be done in the dtor.)

I would still argue for putting this implementation in a helper function rather than forcing all consumers to use the UMA interface directly.

Reply:
The new patch does something along this lines.